src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -140,7 +140,17 @@ public function decode(string $data, string $format, array $context = []): mixed
             }
         }
 
-        // todo: throw an exception if the root node name is not correctly configured (bc)
+        // Validate root node existence and, if configured by the user, its name
```

Comment:
If I get the code right, this does not preserve backward compatibility yet so imho a new option will be needed, default to `false`, trigger a deprecation and throw in 8.0 branch